centos/base/Dockerfile

```diff
@@ -0,0 +1,216 @@
+        # --enable-loadable-sqlite-extensions \
+        # --enable-shared \
+        # --with-system-expat \
+        # --with-system-ffi \
+    && make -j "$(nproc)" \
+    && cd ..
+
+FROM centos:8 AS release
+
+ENV container docker
+
+ARG ST2_VERSION
+ARG ST2CHATOPS_VERSION
+ARG ST2WEB_VERSION
+
+RUN : "${ST2_VERSION:?Docker build argument needs to be set and non-empty.}"
+
+LABEL maintainer="StackStorm <info@stackstorm.com>"
+LABEL com.stackstorm.vendor="StackStorm"
+LABEL com.stackstorm.support="Community"
+LABEL com.stackstorm.version="${ST2_VERSION}"
+LABEL com.stackstorm.name="StackStorm K8s HA CentOS"
+LABEL com.stackstorm.description="Docker CentOS image, optimized to run StackStorm \
+components and core services with Highly Available requirements in Kubernetes environment"
+LABEL com.stackstorm.url="https://stackstorm.com/#product"
+
+
+
+ENV LANG=en_US.UTF-8 \
+    LANGUAGE=en_US:en \
+    LC_ALL=en_US.UTF-8
+
+RUN dnf update -y \
+    && dnf install -y \
+        make \
+        glibc-langpack-en \
+        glibc-locale-source \
+        glibc-langpack-en \
+        glibc-locale-source \ 
+        openssl \
+        epel-release \
+        sudo \
+        wget \
+        git \
+        httpd-tools \
+        systemd \
+        bind \
+        bind-utils \
+        gcc \
+        openldap-devel \
+        gettext \
+        patch \
+        diffutils \
+        zlib-devel \
+    && localedef --no-archive -i en_US -f UTF-8 en_US.UTF-8 \
+    && curl -s https://packagecloud.io/install/repositories/StackStorm/stable/script.rpm.sh | bash \
+    && curl -sL https://rpm.nodesource.com/setup_14.x | bash - \
+    && dnf install -y \
+        st2-${ST2_VERSION} \
+        st2web-${ST2WEB_VERSION} \
+        st2chatops-${ST2CHATOPS_VERSION} \
+        crudini \
+    && dnf clean expire-cache -y \
+    && dnf clean all
+
+
+COPY --from=base /Python-3.7.6 /Python-3.7.6
+
+RUN cd /Python-3.7.6 \
+    && make -j "$(nproc)" altinstall \
+    && cd .. \
+    && rm -rf /Python-3.7.6
+
+ENV LANG=en_US.UTF-8 \
+    LANGUAGE=en_US:en \
+    LC_ALL=en_US.UTF-8 \
+    REQUESTS_CA_BUNDLE=/etc/ssl/certs/ca-bundle.crt
+
+ENV LANG=en_US.UTF-8
+ENV LC_ALL=en_US.UTF-8
+
+# Add custom st2 empty config file, used to override original st2.conf
+COPY conf/st2.docker.conf /etc/st2/
+COPY conf/st2.user.conf /etc/st2/
+# Overrides $ST2_CONF for st2ctl to inject several config files
+COPY conf/st2ctl /etc/default/
+COPY conf/logging.docker.conf /etc/st2/
+
+# Creats the encryption key for the datastore
+RUN mkdir -p /etc/st2/keys \
+    && st2-generate-symmetric-crypto-key --key-path /etc/st2/keys/datastore_key.json \
+    && usermod -a -G st2 st2 && chgrp st2 /etc/st2/keys && chmod o-r /etc/st2/keys \
+    && chgrp st2 /etc/st2/keys/datastore_key.json && chmod o-r /etc/st2/keys/datastore_key.json
+
+# Creates stanley user
+RUN mkdir -p /home/stanley/.ssh && chmod 0700 /home/stanley/.ssh \
+    && ssh-keygen -f /home/stanley/.ssh/stanley_rsa -P "" \
```

Comment:
Just a note that generating SSH key for stanley and bundle it in a Docker image would be a dangerous way.
Though fine for a demo image.

Reply:
good catch, should do this in a separate built process